### DISCUSSION TITLE
How can I use a version of a tool that was installed outside of mise?

### DISCUSSION THREAD

**Discussion by hisaac** [jdx/mise#1768] [Q&A] [answered]
Is it possible to tell mise to use a version of a tool/sdk that was installed by some other means?

A specific example, I'd like to use [JetBrains' `jbr` version of Java](https://github.com/JetBrains/JetBrainsRuntime), but it's not installable via mise's java plugin (or the one used by asdf). It would be nice if I could install the `jbr`, and then point mise to it as the version to use.

**Comment by roele** [comment]
You can also "manually" install tools by extracting them to `~/.local/share/mise/installs/java/<jbr-version>` and then run `mise cache clear`. They will show up as they where installed via mise. This approach can also be used for versions not available for your architecture (e.g. using older Node.js versions via Rosetta on ARM).

Edit: 
For Java a certain folder structure is expected so its a bit more involved that simply extracting the archive but its doable.

```bash
tar xzf jbr_jcef-17.0.10-osx-aarch64-b1207.14.tar.gz
cd jbr_jcef-17.0.10-osx-aarch64-b1207.14
mv Contents/Home/** ./
rm -rf Contents/Home/
ln -s ../ Contents/Home
```

**Reply by jdx** [reply, marked as answer]
also the [`path:` scope](https://mise.jdx.dev/configuration.html#scopes)

**Reply by roele** [reply]
Right, scope `path:` seems to work fine via `mise use java@path:/Users/user/Downloads/jbr_jcef-17.0.10` (assuming the correct folder structure). But does not do anything when used via `mise install ...`.
